src/snapred/backend/data/GroceryService.py

```diff
@@ -190,52 +237,110 @@ def getCloneOfWorkspace(self, name: WorkspaceName, copy: WorkspaceName):
             ws = None
         return ws
 
-    def _updateNeutronCacheFromADS(self, runId: str, useLiteMode: bool):
+    ## FETCH METHODS
+    """
+    The fetch methods orchestrate finding data files, loading them into workspaces,
+    and preserving a cache to prevent re-loading the same data files.
+    """
+
+    def _fetchInstrumentDonor(self, runId: str, useLiteMode: bool) -> WorkspaceName:
+        self._updateInstrumentCacheFromADS(runId, useLiteMode)
+
+        key = self._key(runId, useLiteMode)
+        if self._loadedInstruments.get(key) is not None:
+            return self._loadedInstruments.get(key)
```

Comment:
Could you rewrite this function to have only a single return?

Reply:
Done.  Note that there are at least two coding conventions right now, **the one used by my one-previous development group was "early return".**  I was actually trained to always only have one return in a method, so that kind-of bugged me.